### DISCUSSION TITLE
have Tile card containers and buttons follow HA border-radius theme variable

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [home-assistant/frontend#21357] [Dashboard]
trying to get the Tile card buttons and containers in style with my 0px default border-radius set in my themes, this is a bit of an issue.... I did manage to find some options using card_mod, but, as these elements are all named differently per device the tile cards are controlling, that modding is very complex, and prone to errors:

the tile cards seem to follow the hard-coded 10 px
<img width="636" alt="Scherm­afbeelding 2024-07-10 om 16 42 11" src="https://github.com/home-assistant/frontend/assets/33354141/fc804d3f-b21a-4dbd-bc6d-b12247305d2a">

and Id love to be able to have them follow my theme settings.

currently we need to do something like:

```
      type: tile
      card_mod:
        style:
          hui-card-features $ hui-target-temperature-card-feature $: |
            ha-control-number-buttons {
              --control-number-buttons-border-radius: var(--ha-card-border-radius);
            }

          hui-card-features $:
            hui-climate-preset-modes-card-feature $:
              ha-control-select$: |
                .container {
                  --control-select-border-radius: var(--ha-card-border-radius);
                }
```

and this does not yet set the border-radius on the buttons in:
<img width="560" alt="Scherm­afbeelding 2024-07-10 om 17 18 43" src="https://github.com/home-assistant/frontend/assets/33354141/40191fa3-836d-4bf9-b444-58a8fd97b04f">

this is a FR to have a tile card variable control these border-radius's, so the user can set it in the theme for all Tile card features for light, climate, alarm_panel, slider etc etc

Ofc, if there already is such a variable, please let me know so I can delete this Discussion/FR and apologize for missing that in https://github.com/home-assistant/frontend/blob/dev/src/panels/lovelace/cards/hui-tile-card.ts

**Comment by Mariusthvdb** [comment]
found the button to be moldable like this:
```
      card_mod:
        style:
          hui-card-features $ hui-target-temperature-card-feature $: |
            ha-control-number-buttons {
              --control-number-buttons-border-radius: var(--ha-card-border-radius);
            }

          hui-card-features $:
            hui-climate-preset-modes-card-feature $:
              ha-control-select$: |
                .container {
                  --control-select-border-radius: var(--ha-card-border-radius);
                  --control-select-button-border-radius: var(--ha-card-border-radius);
                }
```

resulting in the desired 0 border-radius on all parts of the interface of the tile:

<img width="520" alt="Scherm­afbeelding 2024-07-11 om 08 39 44" src="https://github.com/home-assistant/frontend/assets/33354141/c6916fca-dbbc-4516-8f22-91f6591f21cc">

and also when hovering:

<img width="519" alt="Scherm­afbeelding 2024-07-11 om 08 42 10" src="https://github.com/home-assistant/frontend/assets/33354141/042e39a7-8d19-43ed-91af-ed37a99ab338">

fan tile is somewhat more involved, using a variable, and a direct element:
```
style:
  hui-card-features $:
    hui-fan-speed-card-feature $:
      ha-control-slider$: |
        .container {
          --control-slider-border-radius: var(--ha-card-border-radius);
        }
        .slider .slider-track-bar {
          border-radius: var(--ha-card-border-radius) !important;
        }
```
to be able to result in

<img width="529" alt="Scherm­afbeelding 2024-07-11 om 09 48 45" src="https://github.com/home-assistant/frontend/assets/33354141/253fe771-595c-4c53-87df-e18c9ef77fa4">

given the complexity of the above modding, the ask remains:
please have these variables available for theming

**Reply by Mariusthvdb** [reply]
Ive made an [overview post in the community for the Tiles on all domains I use](https://community.home-assistant.io/t/how-to-set-border-radius-on-tile-cards/748500/2?u=mariusthvdb), which illustrates the efforts we need to do for such a simple thing, it the elements in the Tile card would simply follow ha-card-border-radius, or if that cant be done, use a themeable variable used throughout.

**Reply by ChristophCaina** [reply]
I would really appreciate, if this would be possible with a simple theme-variable.

Something like
- feature-border
- feature-border-radius

**Comment by Mariusthvdb** [comment]
Testing 2024.8 beta, I just noticed the succes of the overview post has now been obliterated.... the mods no longer kick in and all we get is the default border-radius back

Not sure what has changed. personally I didnt change anything in my themes, so hope/fear it is something on core HA Frontend that was changed

**Reply by Mariusthvdb** [reply]
Could it be this is introduced in ha 2024.8:
```
    feature-border-radius: 0
    control-slider-border-radius: 0
    control-button-border-radius: 0
    control-select-button-border-radius: 0
```

I see them in inspector now (and my memory is blank on those)... so if new, that would be magic.

However, they dont work just yet... at least, I set the above section in my theme, and still all show rounded 18

Hope Paul can give some clarification here.

**Reply by Mariusthvdb** [reply]
nevermind, I had a brainflush.... the variables Are already used in my own mods....

Yet, I wonder why we can not use them in the themes then.

**Reply by ChristophCaina** [reply]
yes, I would really appreciate this option - rather than "card-modding" more cards as already done for 'basic' stuff like state based icon colors... ;)

**Reply by Mariusthvdb** [reply]
for the moment, I am completely lost on the why these no longer function. Dom path seems unchanged, my themes are unchanged, and the mods are unchanged....

---EDIT---
must be this PR that was merged in 2024.8 to adjust Tile card features amongst other things
https://github.com/home-assistant/frontend/pull/21311/files
